### PR TITLE
Container env variables

### PR DESCRIPTION
**Pull request by TPRobots**
Pull request opened by github-pullrequestcreator.

### CODE REVIEW

File: supervisor.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ import (
 	"bufio"
 	"encoding/json"
 	"fmt"
-	docker "github.com/yhat/go-docker"
+	docker "github.com/tmo-trustpilot/go-docker"
```
Comment (paddie) [inline supervisor.go]: it seems you updated the docker remote api client to your own clone of the project. You most likely branched the repo, cloned it to your machine and this import got updated by go-import.

File: supervisor.go [hunk 2]
```diff
@@ -56,6 +56,27 @@ func NewStreamWrapper(su *Supervisor, container *docker.Container, injectedTags
...
 
+	containerInfo, err := su.Cli.InspectContainer(container.Id)
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+
+	tags := make(map[string]string)
+	for k := range containerInfo.Config.Labels {
+		//DOCKERLOCKER.TAG.TEST -> ["DOCKERLOCKER", "TAG", "TEST"]
```
Comment (paddie) [inline supervisor.go]: should this not be `DOCKERLOGGER` instead of `DOCKERLOCKER`?

File: supervisor.go [hunk 3]
```diff
@@ -56,6 +56,27 @@ func NewStreamWrapper(su *Supervisor, container *docker.Container, injectedTags
...
+		labelParts := strings.Split(k, ".")
+		if len(labelParts) < 3 || labelParts[0] != "DOCKERLOCKER" || labelParts[1] != "TAG" {
```
Comment (paddie) [inline supervisor.go]: Same

File: supervisor.go [hunk 4]
```diff
@@ -56,6 +57,21 @@ func NewStreamWrapper(su *Supervisor, container *docker.Container, injectedTags
 	r, w := io.Pipe()
 	r2, w2 := io.Pipe()
 
+	containerInfo, err := su.Cli.InspectContainer(container.Id)
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+	instanceTags := getAllTags(containerInfo.Config.Env)
```
Comment (sqdk) [inline supervisor.go]: I would rename this to containerTags to emphasize that they come from the container and not the instance they are running on.

**Comment by sqdk** [comment]
supervisor.go#L157 is clearly my mistake. Can you remove the line?

**Comment by sqdk** [comment]
I would like that container tags are separated from instance tag in the message as well. It should be possible to extend the StreamWrapper and set instance and container tags seperately and add them seperately under a different name.

I would suggest leaving instance tags as Tags and adding container tags as ContainerTags or something similar.

I would like to prevent that instance tags can be overwritten by mistake.

File: supervisor.go (hunk 1 above)
Comment (tmo-trustpilot) [inline supervisor.go]: This was before I realised I could pull the Env variables off the container config and was trying to use the container labels. The `go-docker` library is missing that field on the struct it deserialises into so I had my own fork with the field on it.

File: supervisor.go (hunk 2 above)
Comment (tmo-trustpilot) [inline supervisor.go]: Blind copy paste :whale:

**Pull request merged**